### DISCUSSION TITLE
Unable to automatically determine debugger corresponding to compiler

### DISCUSSION THREAD

**Discussion by uuuuu-diwu** [microsoft/vscode-cmake-tools#2877] [Q&A] [answered]
[debugger] Unable to automatically determine debugger corresponding to compiler: ~/toolchains/llvm-13/bin/clang

platform： Linux 
compiler：clang 13
launch.json : 

{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "type": "lldb",
            "request": "launch",
            "name": "Launch",
            "program": "${workspaceFolder}/build/main",
            "args": [],
            "cwd": "${workspaceFolder}",
        },
    ]
}

**Comment by bobbrow** [comment, marked as answer]
CMake Tools does not use your `launch.json` to start the debugger.  It generates one based on the information it has and goes through the C/C++ extension instead of the Code LLDB extension.  If you have a launch.json that works, then use the `F5` key to start debugging instead of the debug button in the status bar.  You can also remove the debug button from the status bar with the `cmake.statusbar.advanced` setting if that helps you avoid clicking it.

**Reply by uuuuu-diwu** [reply]
thanks, it works when I delete launch.json
